src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final FruitDao fruitDao;
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(FruitDao fruitDao, OperationStrategy operationStrategy) {
+        this.fruitDao = fruitDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            Fruit fruit = fruitDao.getFruitByName(transaction.getFruit());
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            int balance = handler.executeOperation(fruit.getQuantity(), transaction.getQuantity());
+            fruit.setQuantity(balance);
+            fruitDao.update(fruit);
+        }
+    }
+}
```

Comment:
The class ShopServiceImpl seems to be dependent on both FruitDao and OperationStrategy, which might not align with making services independent as stated in the checklist. Consider calling the services independently in the main() method rather than injecting them into ShopServiceImpl.

Reply:
--